### PR TITLE
Implement Indian numbering system support

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR adds support for the Indian numbering system within the following formatter functions: `fmt_number()`, `fmt_integer()`, and `fmt_currency()`. It's exposed as the `system` argument with two options `"intl"` (international, the default) and `"ind"` (Indian).

With this, the following comparison table was made:

fmt-integer-system

Fixes: https://github.com/rstudio/gt/issues/776

### CODE REVIEW

File: R/format_data.R [hunk 1]
```diff
@@ -2831,3 +2849,111 @@ fmt <- function(data,
...
+
+    integer <- as.character(integer)
+
+  } else if (
+    inherits(integer, "character") &&
+    !grepl("^[0-9]+?$", integer)
+  ) {
+
+    stop(
+      "If `integer` given as a character value, it must only contain numbers."
```
[the remark this answers is not in the document]
Comment (rich-iannone) [reply R/format_data.R]: Changed. Thanks!

File: R/format_data.R [hunk 2]
```diff
@@ -2831,3 +2868,80 @@ fmt <- function(data,
 
   dt_formats_add(data = data, formats = formatter_list)
 }
+
+#' Insert separator marks to an integer to conform to Indian numbering system
+#'
+#' @param integer The integer portion of a numeric value. Should be supplied as
+#'   a string that only contains numeral characters.
```
Comment (jcheng5) [inline R/format_data.R:2875]: Indicate that this needs to be length 1

File: R/utils.R [hunk 3]
```diff
@@ -828,6 +828,95 @@ num_suffix <- function(x,
...
+#' Get a tibble of scaling values and suffixes for the Indian numbering system
+#'
+#' The `num_suffix()` function operates on a vector of numerical values and
+#' returns a tibble where each row represents a scaled value for `x` and the
+#' correct suffix to use during `x`'s character-based formatting.
+#' @noRd
+num_suffix_ind <- function(x,
+                           suffixes = c(NA, "L", "Cr"),
+                           scale_by) {
+
```
Comment (jcheng5) [inline R/utils.R:840]: Check that scale_by is length 1

File: R/utils_formatters.R [hunk 4]
```diff
@@ -302,6 +305,40 @@ format_num_to_str <- function(x,
     x_str[x_str_no_dec] <- paste_right(x_str[x_str_no_dec], dec_mark)
   }
 
+  # Perform modifications to `x_str` values if formatting values to
+  # conform to the Indian numbering system
+  if (system == "ind") {
```
Comment (jcheng5) [inline R/utils_formatters.R:310]: left off here

File: R/utils_formatters.R [hunk 5]
```diff
@@ -256,7 +256,10 @@ format_num_to_str <- function(x,
                               drop_trailing_zeros,
                               drop_trailing_dec_mark,
                               format = "f",
-                              replace_minus_mark = TRUE) {
+                              replace_minus_mark = TRUE,
+                              system = c("intl", "ind")) {
+
```
Comment (jcheng5) [inline R/utils_formatters.R:261]: If `system` is `ind`, hard code `sep_mark` and `dec_mark` here. (Maybe warn if they're not missing and not the hardcoded values?)

File: R/utils_formatters.R [hunk 6]
```diff
@@ -302,6 +305,40 @@ format_num_to_str <- function(x,
...
+    has_decimal <- grepl("\\.", x_str_numeric)
+    is_negative <- grepl("^-", x_str_numeric)
+
+    integer_parts <- x_str_numeric
+
+    if (dec_mark != "") {
+      integer_parts <-
+        strsplit(x_str_numeric, split = dec_mark, fixed = TRUE) %>%
+        vapply(FUN.VALUE = character(1), USE.NAMES = FALSE, `[[`, 1)
+    }
```
Comment (jcheng5) [inline R/utils_formatters.R:323]:
```suggestion
    integer_parts <- sub("\\..*", "", x_str_numeric)
```

File: R/utils_formatters.R [hunk 7]
```diff
@@ -302,6 +305,40 @@ format_num_to_str <- function(x,
...
+        USE.NAMES = FALSE,
+        FUN = insert_seps_ind
+      )
+
+    decimal_str <- rep("", length(x_str_numeric))
+    decimal_str[has_decimal] <-
+      gsub("^.*?(\\..*?)", "\\1", x_str_numeric[has_decimal])
+
+    x_str[!is_inf] <- paste0(integer_parts, decimal_str)
+    x_str[!is_inf][is_negative] <- paste0("-", x_str[!is_inf][is_negative])
```
Comment (jcheng5) [inline R/utils_formatters.R:339]:
```suggestion
    x_str[!is_inf] <- paste0(
      ifelse(is_negative, "-", ""),
      integer_parts,
      decimal_str
    )
```

File: R/utils_formatters.R (hunk 6 above)
Comment (rich-iannone) [reply R/utils_formatters.R:323]: Done

File: R/utils_formatters.R [hunk 8]
```diff
@@ -302,6 +305,40 @@ format_num_to_str <- function(x,
...
+      gsub("(,|-)", "", .) %>%
+      vapply(
+        FUN.VALUE = character(1),
+        USE.NAMES = FALSE,
+        FUN = insert_seps_ind
+      )
+
+    decimal_str <- rep("", length(x_str_numeric))
+    decimal_str[has_decimal] <-
+      gsub("^.*?(\\..*?)", "\\1", x_str_numeric[has_decimal])
```
[the remark this answers is not in the document]
Comment (rich-iannone) [reply R/utils_formatters.R:336]: Done.

**Pull request merged**